src/clonemap.rs

```diff
@@ -0,0 +1,717 @@
+//! The [`CloneConMap`][crate::CloneConMap] type and its helpers.
+
+use std::borrow::Borrow;
+use std::collections::hash_map::RandomState;
+use std::fmt::{Debug, Formatter, Result as FmtResult};
+use std::hash::{BuildHasher, Hash};
+use std::iter::FromIterator;
+use std::marker::PhantomData;
+
+use crate::existing_or_new::ExistingOrNew;
+use crate::raw::config::Config;
+use crate::raw::{self, Raw};
+
+#[derive(Clone)]
+struct CloneMapPayload<K, V>((K, V));
+
+impl<K, V> Borrow<K> for CloneMapPayload<K, V> {
+    fn borrow(&self) -> &K {
+        let &(ref k, _) = &self.0;
+        k
+    }
+}
+
+struct CloneMapConfig<K, V>(PhantomData<(K, V)>);
+
+impl<K, V> Config for CloneMapConfig<K, V>
+where
+    K: Clone + Hash + Eq + 'static,
+    V: Clone + 'static,
+{
+    type Payload = CloneMapPayload<K, V>;
+    type Key = K;
+}
+
+/// The iterator of the [`CloneConMap`].
+///
+/// See the [`iter`][CloneConMap::iter] method for details.
+pub struct Iter<'a, K, V, S>
+where
+    K: Clone + Hash + Eq + 'static,
+    V: Clone + 'static,
+{
+    inner: raw::iterator::Iter<'a, CloneMapConfig<K, V>, S>,
+}
+
+impl<'a, K, V, S> Iterator for Iter<'a, K, V, S>
+where
+    K: Clone + Hash + Eq + 'static,
+    V: Clone + 'static,
+{
+    type Item = (K, V);
+    fn next(&mut self) -> Option<(K, V)> {
+        self.inner.next().map(|p| (p.0).clone())
+    }
+}
+
+/// A concurrent map that clones its elements.
+///
+/// This flavour stores the data as `(K, V)` tuples; it clones
+/// independently both the elements (to be indended as the key and the
```

Comment:
Is `indended` a word I don't know (possible, English is not my first language) or some typo?

Reply:
Typo! fixed...